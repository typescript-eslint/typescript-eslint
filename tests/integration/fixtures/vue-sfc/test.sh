#!/bin/bash
set -exuo pipefail

# Generate the package.json to use
node /usr/utils/generate-package-json.js

# Install dependencies
npm install

# Use the local volumes for our own packages
npm install $(npm pack /usr/types | tail -1)
npm install $(npm pack /usr/visitor-keys | tail -1)
npm install $(npm pack /usr/scope-manager | tail -1)
npm install $(npm pack /usr/typescript-estree | tail -1)
npm install $(npm pack /usr/parser | tail -1)
npm install $(npm pack /usr/experimental-utils | tail -1)
npm install $(npm pack /usr/eslint-plugin | tail -1)

# Install the latest versions of dependencies (this may break us occassionally, but it's probably good to get that feedback early)
npm install vue-eslint-parser@latest eslint-plugin-vue@latest
# Install the latest some other vue utilities
npm install vuex@latest vue-property-decorator@latest

# Run the linting
# (the "|| true" helps make sure that we run our tests on failed linting runs as well)
npx eslint --format json --output-file /usr/lint-output.json --config /usr/linked/.eslintrc.js /usr/linked/**/*.vue || true

# Run our assertions against the linting output
npx jest /usr/test.js --snapshotResolver=/usr/utils/jest-snapshot-resolver.js
