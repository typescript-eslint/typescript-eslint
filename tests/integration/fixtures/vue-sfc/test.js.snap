// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should produce the expected lint ouput 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "/usr/linked/Hello.vue",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 25,
        "endColumn": 28,
        "endLine": 33,
        "line": 33,
        "message": "Unexpected any. Specify a different type.",
        "messageId": "unexpectedAny",
        "nodeType": "TSAnyKeyword",
        "ruleId": "@typescript-eslint/no-explicit-any",
        "severity": 2,
        "suggestions": Array [
          Object {
            "desc": "Use \`unknown\` instead, this will force you to explicitly, and safely assert the type is correct.",
            "fix": Object {
              "range": Array [
                648,
                651,
              ],
              "text": "unknown",
            },
            "messageId": "suggestUnknown",
          },
          Object {
            "desc": "Use \`never\` instead, this is useful when instantiating generic type parameters that you don't need to know the type of.",
            "fix": Object {
              "range": Array [
                648,
                651,
              ],
              "text": "never",
            },
            "messageId": "suggestNever",
          },
        ],
      },
    ],
    "source": "<!-- Hello.vue -->
<template>
  <div>
    <div>
      <!-- !!!!! expected error !!!!! -->
      Hello {{ (name as any) }}{{ exclamationMarks }}
    </div>
    <button @click=\\"decrement\\">-</button>
    <button @click=\\"increment\\">+</button>
  </div>
</template>

<script lang=\\"ts\\">
import Vue from 'vue';
export default Vue.extend({
  props: ['name', 'initialEnthusiasm'],
  data() {
    return {
      enthusiasm: this.initialEnthusiasm,
    };
  },
  methods: {
    increment() {
      this.enthusiasm++;
    },
    decrement() {
      if (this.enthusiasm > 1) {
        this.enthusiasm--;
      }
    },
  },
  computed: {
    exclamationMarks(): any {
      // !!!!! expected error !!!!!
      return Array(this.enthusiasm + 1).join('!');
    },
  },
});
</script>
",
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
  Object {
    "errorCount": 0,
    "filePath": "/usr/linked/Utility.vue",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [],
    "output": "<script lang=\\"ts\\">
/* https://github.com/typescript-eslint/typescript-eslint/issues/2591 */
/* eslint-disable class-methods-use-this */
export default class Utility {
  get a() {
    const list: Array<string> = [];
    return list;
  }

  get b() {
    const a = this.a;
    return a;
  }
}
</script>
",
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
  Object {
    "errorCount": 0,
    "filePath": "/usr/linked/World.vue",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;
